src/App.jsx

```diff
@@ -1,5 +1,81 @@
-function App() {
-  return <div className="App"></div>;
+import * as yup from "yup";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { useForm } from "react-hook-form";
+
+const schema = yup.object().shape({
+  email: yup.string().email().required(),
+  password: yup.string().min(8).max(32).required(),
+});
+
+// login component
+function Login() {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm({
+    resolver: yupResolver(schema),
+  });
+
+  const onSubmitHandler = (data) => {
+    console.log({ data });
+    reset();
+  };
```

Comment:
Perfect! Now we have the validation all set.
Can you now refactor this and move it into a new file `use-login.ts`

This file will have all the logic and return functions and data to the component

[Read about separation of concerns in react](https://www.freecodecamp.org/news/separation-of-concerns-react-container-and-presentational-components/)